### PR TITLE
[Support] Return id for  importannotations 

### PR DESCRIPTION
**Pull request by kmathew-pdftron**

### CODE REVIEW

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 1]
```diff
@@ -3132,12 +3134,56 @@ public void importAnnotations(String xfdf, boolean replace) throws PDFNetExcepti
...
+            Obj result = obj.findObj(key);
+            if (result != null && result.isNumber()) {
+                double number = result.getNumber();
+                return (int) number;
+            }
+        }
+        return null;
+    }
+
+    private static WritableArray getAnnotationsFromXFDF(String xfdf) throws PDFNetException{
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3163]: cleanup import and code formatting

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -3132,12 +3134,56 @@ public void importAnnotations(String xfdf, boolean replace) throws PDFNetExcepti
...
                 pdfViewCtrl.update(true);
+
             } finally {
                 if (shouldUnlock) {
                     pdfViewCtrl.docUnlock();
                 }
             }
         }
+        try {
+            return getAnnotationsFromXFDF(xfdf);
```
Comment (sgong-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3145]: you only want to return this when the merge is successful, so you want to have this line right after pdfviewctrl.update(true)

**Review by sgong-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 2 above)
Comment (kmathew-pdftron) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3145]: what about for collaboration mode?

Comment (sgong-pdftron) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3145]: return this after "mCollabManager.importAnnotations(xfdf, false);"

Comment (sgong-pdftron) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3145]: you want to maintain the code readability, so return when needed is important

File: API.md [hunk 3]
```diff
@@ -2400,11 +2400,19 @@ Name | Type | Description
 xfdf | string | annotation string in XFDF format for import
 replace | boolean | whether to replace existing form and annotation data with those imported from the XFDF string (Android only) 
 
-Returns a Promise.
+Promise Parameters:
```
Comment (sgong-pdftron) [inline API.md:2403]: `Returns a Promise.` should not be removed

File: src/AnnotOptions/AnnotOptions.ts [hunk 4]
```diff
@@ -8,6 +8,11 @@ export interface Annotation {
     screenRect?: Rect;
 }
 
+export interface ImportAnnotations {
```
Comment (sgong-pdftron) [inline src/AnnotOptions/AnnotOptions.ts:11]: should be using Annotation interface instead

**Review by sgong-pdftron — APPROVED** [review]

File: ios/RNTPTDocumentView.m [hunk 5]
```diff
@@ -1294,7 +1294,7 @@ - (NSString *)exportAnnotationsWithOptions:(NSDictionary *)options
     return nil;
 }
 
-- (void)importAnnotations:(NSString *)xfdfString
+- (NSMutableArray<NSDictionary *> *)importAnnotations:(NSString *)xfdfString
```
Comment (JamieDass) [inline ios/RNTPTDocumentView.m:1297]: General suggestion to return the immutable version.
Should also specify `nullable` if the method can return `nil`.
```suggestion
- (nullable NSArray<NSDictionary *> *)importAnnotations:(NSString *)xfdfString
```

File: ios/RNTPTDocumentViewManager.h [hunk 6]
```diff
@@ -29,7 +29,7 @@
 
 - (NSString *)exportAnnotationsForDocumentViewTag:(NSNumber *)tag options:(NSDictionary *)options;
 
-- (void)importAnnotationsForDocumentViewTag:(NSNumber *)tag xfdf:(NSString *)xfdfString replace:(BOOL)replace;
+- (NSMutableArray<NSDictionary *> *)importAnnotationsForDocumentViewTag:(NSNumber *)tag xfdf:(NSString *)xfdfString replace:(BOOL)replace;
```
Comment (JamieDass) [inline ios/RNTPTDocumentViewManager.h:32]:
```suggestion
- (nullable NSArray<NSDictionary *> *)importAnnotationsForDocumentViewTag:(NSNumber *)tag xfdf:(NSString *)xfdfString replace:(BOOL)replace;
```

**Review by JamieDass — CHANGES_REQUESTED** [review]

File: ios/RNTPTDocumentView.m [hunk 7]
```diff
@@ -1317,6 +1317,36 @@ - (void)importAnnotations:(NSString *)xfdfString
...
+    return [self getAnnotationFromXFDF:xfdfString];
+
+}
+
+-(NSMutableArray<NSDictionary *> *)getAnnotationFromXFDF:(NSString *)xfdfString
+{
+    NSMutableArray<NSDictionary *> *annotations = [[NSMutableArray alloc] init];
+    PTFDFDoc *fdfDoc = [PTFDFDoc CreateFromXFDF:xfdfString];
+    PTObj *fdf = [fdfDoc GetFDF];
+    if (fdf != nil) {
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:1330]: The Obj-C PDFNet wrapper always returns a value, and you have to check the `IsValid` method instead:
```suggestion
    if ([fdf IsValid]) {
```

File: ios/RNTPTDocumentView.m [hunk 8]
```diff
@@ -1317,6 +1317,36 @@ - (void)importAnnotations:(NSString *)xfdfString
...
+        PTObj *annots = [fdf FindObj:@"Annots"];
+        if (annots != nil && [annots IsArray]) {
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:1332]:
```suggestion
        if ([annots IsValid] && [annots IsArray]) {
```

File: ios/RNTPTDocumentView.m [hunk 9]
```diff
@@ -1317,6 +1317,36 @@ - (void)importAnnotations:(NSString *)xfdfString
...
+            long size = [annots Size];
+            for (int i = 0; i < size; i++) {
+                PTObj *annotObj = [annots GetAt:i];
+
+                if (annotObj != nil) {
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:1337]:
```suggestion
                if ([annotObj IsValid]) {
```

File: ios/RNTPTDocumentView.m [hunk 10]
```diff
@@ -1317,6 +1317,36 @@ - (void)importAnnotations:(NSString *)xfdfString
...
     }
+    
+    return [self getAnnotationFromXFDF:xfdfString];
+
+}
+
+-(NSMutableArray<NSDictionary *> *)getAnnotationFromXFDF:(NSString *)xfdfString
+{
+    NSMutableArray<NSDictionary *> *annotations = [[NSMutableArray alloc] init];
+    PTFDFDoc *fdfDoc = [PTFDFDoc CreateFromXFDF:xfdfString];
```
Comment (dluco) [inline ios/RNTPTDocumentView.m:1328]: Should wrap the rest of this method in a try-catch to keep any exceptions local:
```suggestion
    @try {
        PTFDFDoc *fdfDoc = [PTFDFDoc CreateFromXFDF:xfdfString];
```

**Review by dluco — CHANGES_REQUESTED** [review]

**Review by JamieDass — APPROVED** [review]

**Review by dluco — APPROVED** [review]

**Pull request merged**